File: voyageone-static/2.0/develop/modules/cms/models/SimpleMappingBean.js
define([
    'modules/cms/enums/MappingTypes'
], function (MappingTypes) {

    /**
     * @name MappingBean
     * @class
     * @description
     * 各类型 MappingBean 的父类, 位于 com.voyageone.cms.service.bean.MappingBean
     */

    /**
     * com.voyageone.cms.service.bean.SimpleMappingBean
     * @constructor
     * @extends {MappingBean}
     */
    function SimpleMappingBean() {
        /**
         * 平台属性 ID
         * @type {string}
         */
        this.platformPropId = null;

        this.mappingType = MappingTypes.SIMPLE_MAPPING;
        /**
         * 表达式
         * @type {RuleExpression}
         */
        this.expression = null;
    }

    return SimpleMappingBean;

});

File: voyageone-service2/voyageone-service2-model/src/main/java/com/voyageone/service/bean/cms/SimpleMappingBean.java
package com.voyageone.service.bean.cms;

import com.voyageone.ims.rule_expression.RuleExpression;

/**
 * Created by Leo on 15-12-9.
 */
public class SimpleMappingBean extends MappingBean {
    private RuleExpression expression;

    public SimpleMappingBean() {
        mappingType = MAPPING_SIMPLE;
    }

    public SimpleMappingBean(String platformPropId, RuleExpression expression) {
        this();
        this.expression = expression;
        this.platformPropId = platformPropId;
    }

    public RuleExpression getExpression() {
        return expression;
    }

    public void setExpression(RuleExpression expression) {
        this.expression = expression;
    }
}
